From 151a02a6e996c3c64f8af8a0b421a26ffd6a6bd5 Mon Sep 17 00:00:00 2001
From: Prabhu <prabhu.sundararaj@freescale.com>
Date: Wed, 26 Aug 2015 16:45:47 -0500
Subject: [PATCH] Reuse the resurface only when window is available

There are cases where the window is destroyed but the static
variable __singleWindowSurface may not get updated. So
additional check whether window is not destroyed before deciding
to reuse the surface.

Signed-off-by: Prabhu <prabhu.sundararaj@freescale.com>
---
 src/plugins/platforms/eglfs/qeglfswindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/eglfs/qeglfswindow.cpp b/src/plugins/platforms/eglfs/qeglfswindow.cpp
index 2c05979..c59d795 100644
--- a/src/plugins/platforms/eglfs/qeglfswindow.cpp
+++ b/src/plugins/platforms/eglfs/qeglfswindow.cpp
@@ -69,7 +69,7 @@ void QEglFSWindow::create()
     m_flags = Created;
 
     static EGLSurface __singleWindowSurface;
-    if(QEglFSHooks::hooks() && ! QEglFSHooks::hooks()->hasCapability(QPlatformIntegration::MultipleWindows) && (__singleWindowSurface)) {
+    if(QEglFSHooks::hooks() && ! QEglFSHooks::hooks()->hasCapability(QPlatformIntegration::MultipleWindows) && (__singleWindowSurface) && m_window != 0) {
         m_surface = __singleWindowSurface;
 #ifdef QEGL_EXTRA_DEBUG
         qWarning("Surface recreate request, re-using %x\n", m_surface);
-- 
2.4.5

